libensemble/generators.py

```diff
@@ -95,12 +102,45 @@ class LibensembleGenerator(Generator):
     """
 
     def __init__(
-        self, History: npt.NDArray = [], persis_info: dict = {}, gen_specs: dict = {}, libE_info: dict = {}, **kwargs
+        self,
+        variables: dict,
+        objectives: dict = {},
+        History: npt.NDArray = [],
+        persis_info: dict = {},
+        gen_specs: dict = {},
+        libE_info: dict = {},
+        **kwargs,
     ):
+        self.variables = variables
+        self.objectives = objectives
+        self.History = History
         self.gen_specs = gen_specs
+        self.libE_info = libE_info
+
+        self.variables_mapping = kwargs.get("variables_mapping", {})
+
+        self._internal_variable = "x"  # need to figure these out dynamically
+        self._internal_objective = "f"
+
+        if self.variables:
+
+            self.n = len(self.variables)
+            # build our own lb and ub
+            if "lb" not in kwargs and "ub" not in kwargs:
```

Comment:
Have you tested if these are in kargs?

Reply:
If I forget to comment out my lb/ub line I get:
...persistent_aposmm.py", line 326, in update_local_H_after_receiving
    local_H[name][Work["libE_info"]["H_rows"]] = calc_in[name]
ValueError: shape mismatch: value array of shape (4,6) could not be broadcast to indexing result of shape (4,3)

Reply:
That was with our branches merged.